### PR TITLE
CSharpException Homework is completed

### PR DESCRIPTION
**Pull request by NadzeyaKiz**
Completed

### CODE REVIEW

File: Code/CSharpException/Triangle.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using System.Threading.Tasks;
+
+
+namespace CSharpException
+{
+    public class Triangle
+    {
+        public Triangle(double sideA, double sideB, double sideC) 
+        {
+            if (sideA <= 0 || sideB <= 0 || sideC <= 0)
```
Comment (MariaEroshevich) [inline Code/CSharpException/Triangle.cs]: better to create separate method to check the sides of the triangle.

File: Code/CSharpException/Triangle.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                throw new ArgumentException("Side value cannot be equal to or less than zero");
+            }
+            if (sideA + sideB <= sideC || sideA + sideC <= sideB || sideB + sideC <= sideA)
+            {
+                throw new ArgumentException("The sum of two sides must be greater than the third side. Please enter a valid values of the triangle sides.");
+            }
+            SideA = sideA;
+            SideB = sideB;
+            SideC = sideC;
+        }
```
Comment (MariaEroshevich) [inline Code/CSharpException/Triangle.cs:21]: add spaces between constructor and all properties.

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: Code/CSharpException/Triangle.cs (hunk 1 above)
Comment (NadzeyaKiz) [reply Code/CSharpException/Triangle.cs]: Done. Two separate methods are created.

File: Code/CSharpException/Triangle.cs (hunk 2 above)
Comment (NadzeyaKiz) [reply Code/CSharpException/Triangle.cs:21]: done

File: Code/CSharpException/Triangle.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            Console.WriteLine($"The sideA of the triangle is: {SideA}");
+            Console.WriteLine($"The sideB of the triangle is: {SideB}");
+            Console.WriteLine($"The sideC of the triangle is: {SideC}");
+        }
+
+        public bool ValidateSidesNotNull (double sideA, double sideB, double sideC)
+        {
+            if (sideA <= 0 || sideB <= 0 || sideC <= 0)
+            {
+                return false;
```
Comment (MariaEroshevich) [inline Code/CSharpException/Triangle.cs]: better to throw exception here instead of throw it in constructor.

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: Code/CSharpException/Triangle.cs (hunk 3 above)
Comment (NadzeyaKiz) [reply Code/CSharpException/Triangle.cs]: done

File: Code/CSharpException/Triangle.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        public Triangle(double sideA, double sideB, double sideC) 
+        {
+            ValidateSidesNotNull(sideA, sideB, sideC);
+            ValidateSidesSummOfSides(sideA, sideB, sideC);
+                
+            SideA = sideA;
+            SideB = sideB;
+            SideC = sideC;
+
```
Comment (MariaEroshevich) [inline Code/CSharpException/Triangle.cs:20]: remove space

**Review by MariaEroshevich — APPROVED** [review]

File: Code/CSharpException/Triangle.cs (hunk 4 above)
Comment (NadzeyaKiz) [reply Code/CSharpException/Triangle.cs:20]: done

**Pull request merged**